proselint/tools.py

```diff
@@ -160,7 +160,7 @@ def __init__(self, source: str | Path, content: str | None = None) -> None:
 
         self.source = source
 
-        if content:
+        if content is not None:
```

Comment:
It strikes me that this is probably the wrong place to make a change. The broken method is `_read`, in its flawed assumption that `self.content` being empty means there must be a source available to read.

Since GitHub doesn't let me propose changes on unchanged regions, I would suggest something like
```py
            if self.content or isinstance(self.source, str):
```
in place of line 192. This should also remove the need for `cast` in `_read`, but I do not have the opportunity to test that currently.